README.md

```diff
@@ -2,31 +2,242 @@
+        this.footballService = footballService;
+    }
+
+    @PostMapping("/football-team")
+    public String addFootballTeam(@RequestBody FootballTeamDTO footballTeam) {
+        return footballService.addFootballTeam(footballTeam);
+    }
+
+    @GetMapping("/football-team/{teamName}")
+    public FootballTeamDTO getFootballTeam(@PathVariable String teamName) {
+        return footballService.getFootballTeam(teamName);
+    }
+}
+```
+
+```java
+// Service
+
+package com.example.springwebdemo.service;
+
+import com.example.springwebdemo.dto.FootballTeamDTO;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class FootballService {
+    List<FootballTeamDTO> footballTeams;
+
+    public FootballService() {
+        footballTeams = new ArrayList<>();
+    }
+
+    public String addFootballTeam(FootballTeamDTO footballTeam) {
+        footballTeams.add(footballTeam);
+        return String.format("%s has been added!", footballTeam.getTeamName());
+    }
+
+    public FootballTeamDTO getFootballTeam(String teamName) {
+        Optional<FootballTeamDTO> footballTeamDTO = footballTeams.stream()
+                .filter(footballTeam -> footballTeam.getTeamName().equals(teamName))
+                .findAny();
+        
+        // If the team does not exist, throw a NoSuchElementException
+        return footballTeamDTO.orElseThrow();
+    }
+}
+```
+
+Let's break down this code a little bit.
+
+We want to be able to accomplish two main goals with this REST API:
+
+- Add a football team.
+- Get a football team by its team name.
+
+We'll use a `List` data structure within the service class to keep track of the
+football teams. This will also make it easy to add teams and get the teams from
+the list.
+
+But let's look at the `FootballController` class, as we have introduced some new
+annotations.
+
+### @PostMapping
+
+If we want to add a new football team to our application, then we need to use the
+`@PostMapping` annotation. The `@PostMapping` in shorthand for the
+`@RequestMapping(method=RequestMethod.POST)` annotation, which will handle mapping
+HTTP POST requests.
+
+### @RequestBody
+
+This annotation allows the method to accept JSON formatted client data and
+convert it to a Java object. For example, the JSON object mentioned above would
+be mapped to a `FootballTeamDTO` object with the attributes set to the values in
+the JSON.
+
+### Get Method
+
+We'll use the `@GetMapping` and `@PathVariable` annotation that we used before to
+have our application process the GET request. The only difference here is we'll
+return a DTO instead of a `String`. When we do this, it will serialize the data
+into a JSON object to return.
+
+## Passing in a JSON
+
+Let's run the application and see what happens. We will first need to add a
+football team to the application.
+
+Open up Postman and under the "Headers" tab, add "Content-Type":"application/json"
```

Comment:
I believe Postman will automatically set the Content-Type to application/json if you choose JSON from the dropdown in the body tab, too.

Reply:
I didn't know that! Good to know!